server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -1077,6 +1079,10 @@ record PluginServiceInstances(
             );
         }
 
+        getSinglePlugin(InferenceProviderPlugin.class).ifPresent(plugin -> {
```

Comment:
I wonder if there's a better way of binding the InferenceProvider into the TransportBulkAction. This does the trick though.